src/msg/async/PosixStack.h

```diff
@@ -48,14 +48,13 @@ class PosixNetworkStack : public NetworkStack {
       return -1;
     return coreids[id % coreids.size()];
   }
-  virtual void spawn_workers(std::vector<std::function<void ()>> &funcs) override {
-    for (unsigned i = threads.size(); i < funcs.size(); ++i)
-      threads.emplace_back(std::thread(std::move(funcs[i])));
+  virtual void spawn_worker(unsigned i, std::function<void ()> &&func) override {
+    threads.resize(i+1);
+    threads[i] = std::move(std::thread(func));
```

Comment:
Currently it's ok to emplace_back. but we may reenable smart thread spawn. So "i" maybe rollback to smaller